include/rmm/cuda_stream.hpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/detail/error.hpp>
+
+#include <cuda_runtime_api.h>
+
+#include <memory>
+
+namespace rmm {
+
+/**
+ * @brief Owning wrapper for a CUDA stream.
+ *
+ * Provides RAII lifetime semantics for a CUDA stream.
+ */
+class cuda_stream {
+ public:
+  /**
+   * @brief Move constructor (default)
+   *
+   * A moved-from cuda_stream is invalid and it is Undefined Behavior to call methods that access
+   * the owned stream.
+   */
+  cuda_stream(cuda_stream&&) = default;
+  /**
+   * @brief Move copy assignment operator (default)
+   *
+   * A moved-from cuda_stream is invalid and it is Undefined Behavior to call methods that access
+   * the owned stream.
+   */
+  cuda_stream& operator=(cuda_stream&&) = default;
```

Comment:
Previously the copy ctor/assignment were explicitly deleted, and *you* told me that was not necessary since the underlying storage (`unique_ptr`) would cause them to be deleted.   I don't think I should default them since they are deleted...  But maybe I should explicitly delete them.

Reply:
Added back the explicitly deleted copy functions.

Reply:
> you told me that was not necessary since the underlying storage (unique_ptr) would cause them to be deleted. I don't think I should default them since they are deleted... But maybe I should explicitly delete them.

I said they don't need to be explicitly deleted and could be defaulted. It doesn't really matter which you choose, so long as it's explicit. The general rule is that if you delete/default one, you should do the rest.